hangman.py

```diff
@@ -113,19 +145,54 @@ def hangman(secret_word):
     
     Follows the other limitations detailed in the problem write-up.
     '''
```

Comment:
This function is too long. Please - reduce it to have 30 lines maximum (try to extract some functions)